WooCommerce/Classes/ViewRelated/Coupons/Coupon+String.swift

```diff
@@ -0,0 +1,29 @@
+import Foundation
+import Yosemite
+
+extension Coupon {
```

Comment:
This code was previously inside the Coupon details ViewModel, but since we now need this logic in the Coupon list also, I decided to go with code reuse and made the share message something generated from within the Coupon through this extension, not sure if this would be an acceptable way to do that using Swift tho.